promising/logging_utils.py

```
@@ -0,0 +1,119 @@
+import logging
+from collections.abc import Iterable
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from promising.promising_context import PromisingContext
+
+
+class PromisingHierarchyLogger:
+    def __init__(self, *, logger: logging.Logger | None = None, level: int):
```

Ruff (0.15.10)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)